execution/inc/leatherman/execution/option_set.hpp

```diff
@@ -0,0 +1,228 @@
+/**
+ * @file
+ * Declares utility type for passing a set of options.
+ */
+#pragma once
+
+#include <numeric>
+#include <string>
+#include <initializer_list>
+
+namespace leatherman { namespace execution {
```

Comment:
option_set was previously in the util namespace, as it is a general-purpose utility. I'd probably keep it there.